### PR TITLE
use Token type for hex error, draft re #1802

### PR DESCRIPTION
**Pull request by ghost**
Continuing the work done so far by other folks within #1802.

The suggestion here is to

- [x] Provide a working draft skeleton (one hex related error message)
- [x] Extend the draft with several messages (e.g. `IllegalHexOdd` etc.)
- [ ] ~Provide a working draft where the Parser handles Hex~
- [ ] ~Extend the draft with several messages / error-handlers~

The above flow ensures that the hex-error-reporting is enhanced soon (users benefit).

This works is not for nothing, as it lays the foundation for a follow-up refactoring (to do things right, e.g. within the Parser itself).

If the refactoring fails, for any reason, the users will still have the hex-error-reporting based on the Scanner/Token workaround.

### CODE REVIEW

**Comment by ghost** [comment]
@axic , is there any way to suppress this codecov bot? Going totally on my nerves with its verbosity.

**Comment by ghost** [comment]
@chriseth , @axic - Leaving commits un-purged for review, to showcase the deficits of the code better.

The existent code does not catch the `hex1234` (missing opening quote)correctly. I did not correct this, as I got a bad feeling.

The other usual error cases lead to the Error: `Expected even number of hex-nibbles within double-quotes.`

Result: Those commits fix #1802 quick & easy, with slight non-breaking changes, and should be merged.

The next step would be to rework (= refactor and slight redesign) the code, thus hex is handled by other layers, as suggested within https://github.com/ethereum/solidity/issues/1802#issuecomment-391543993.

(I will open a new issue for this)

**Comment by ghost** [comment]
squashed them into one commit.

what is this `build_x86_mac` failure about?

**Comment by erak** [comment]
@lazaridiscom The build environment on Circle was updated last week. We increased the XCode version in our config accordingly: ethereum/solidity#5138. Rebasing your branch onto develop should fix the issue.

File: libsolidity/parsing/Scanner.cpp [hunk 1]
```diff
@@ -611,8 +611,8 @@ void Scanner::scanToken()
 					// Special quoted hex string must follow
 					if (m_char == '"' || m_char == '\'')
 						token = scanHexString();
-					else
-						token = Token::Illegal;
+					else
```
Comment (chriseth) [inline libsolidity/parsing/Scanner.cpp]: trailing space

File: libsolidity/parsing/Token.h [hunk 2]
```diff
@@ -261,6 +261,8 @@ namespace solidity
 	\
 	/* Illegal token - not able to scan. */                            \
 	T(Illegal, "ILLEGAL", 0)                                           \
+    /* Illegal hex token */                                            \
```
Comment (chriseth) [inline libsolidity/parsing/Token.h]: please use tabs

**Review by chriseth — CHANGES_REQUESTED** [review]
Please rebase and fix syntax problems. Apart from that, very nice change!

**Comment by ghost** [comment]
Should be 'k now.

**Review by chriseth — APPROVED** [review]

**Pull request merged**